### PR TITLE
Simplified All Formatting Checks

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
- changed all formatting checks to format everything all the time. This should resolve some of the wierd "no idea why this formats on my branch" issues we've seen in PR's recently
- added formatting for bazel build files
- both `clang-format` and the `bazel buildifier` are downloaded through bazel
- formatting the entire repo (`clang-format` and `bazel buildifier`) takes just 3.57 seconds on my dual core machine
- this does significantly increase the CI time for formatting checking, but we can speed that up in the future by downloading just clang-format instead of all of llvm, and besides, the formatting check is usually not the blocking factor.

### Testing Done
Currently performing testing, but need a PR open to do so.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/external/clang.BUILD [hunk 1]
```diff
@@ -63,3 +63,8 @@ filegroup(
         "lib/libunwind.a",
     ],
 )
+
+sh_binary(
```
Comment (MathewMacDougall) [inline src/external/clang.BUILD:67]: Why is this needed? So we can run formatting as a bazel target?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: src/external/clang.BUILD (hunk 1 above)
Comment (garethellis0) [reply src/external/clang.BUILD:67]: I thought this was a bit neater then committing the binary directly, but formatting is not a bazel target, so this is not required, we could go back to committing the binary in.

File: src/firmware/main/primitives/accurate_shoot.c [hunk 2]
```diff
@@ -23,7 +23,7 @@
 static float destination[3], major_vec[2], minor_vec[2];
 // Only need two data points to form major axis vector.
 
-primitive_params_t *global_params;
+primitive_params_t* global_params;
```
Comment (jonathanlew) [inline src/firmware/main/primitives/accurate_shoot.c:26]: Hm... I like it the other way around `type *var` @MathewMacDougall @akhilveeraghanta what do you think?

**Review by jonathanlew — COMMENTED** [review]
Good, just a personal preference comment

File: src/firmware/main/primitives/accurate_shoot.c (hunk 2 above)
Comment (garethellis0) [reply src/firmware/main/primitives/accurate_shoot.c:26]: So this is a formatting question, so somewhat out of scope here, but I strongly prefer this. If you were to ask what the type of `global_params` is, it's a "primitive_params_t _pointer_", so I think this formatting helps convey the point that the pointer is part of the type, not the name.

**Review by jsn0716 — COMMENTED** [review]
I'm not very familiar with CI / formatting scripts, but from what I could understand it looks good to me

File: src/firmware/main/primitives/accurate_shoot.c (hunk 2 above)
Comment (jonathanlew) [reply src/firmware/main/primitives/accurate_shoot.c:26]: Yeah I see

**Review by jonathanlew — APPROVED** [review]
:+1:

**Review by akhilveeraghanta — APPROVED** [review]
LG:tm: :tada:

**Review by MathewMacDougall — APPROVED** [review]
